### ISSUE TITLE
Remove dom-purify in favor of sanitize-section-content

### ISSUE THREAD

**Issue by blowery** [Automattic/wp-calypso#25622]
from @DanReyLop on dom-purify usage:

> Our use case seems that it could be perfectly covered by sanitize-section-content.js. That file seems to serve the same purpose as dompurify. I'll look at this when I have time, if you don't mind leaving this open for a few days more.

See https://github.com/Automattic/wp-calypso/pull/25505#issuecomment-398404167

**Comment by flootr** [comment]
The main difference between `sanitizeHtml` (using dompurify) and `sanitize-section-content` is what tags and attributes are allowed. `sanitizeHtml` [is more restrictive](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/woocommerce-services/lib/utils/sanitize-html.js#L11-L12). To keep it the same as is right now we'd need to make `sanitize-section-content` configurable. I'm not really sure though if that's necessary or if what `sanitize-section-content` evaluates by default is sufficient as well.

@DanReyLop any thoughts?

**Comment by DanReyLop** [comment]
@flootr Yeah, making `sanitize-section-content` configurable would be the ideal solution. `dompurify` is more complex that we need (it parses MathML for example), and we would also need some logic on top of `dompurify` to match what `sanitize-section-content` is doing (rewriting links from `http` to `https` or rewriting `<h1>` to `<h3>`, for example).

I have a half-baked branch to make `sanitize-section-content` configurable, I'll finish it when I have some down time (it's not a huge priority).

> I'm not really sure though if that's necessary or if what sanitize-section-content evaluates by default is sufficient as well.

`sanitize-section-content` is way too permisive for our use case. Our HTML strings will be used as titles/descriptions for form fields, so we just want to allow basic formatting tags like `<b>`, `<i>` and links.
